app/src/main/java/com/example/drawingApp/DialogUtility.kt

```diff
@@ -0,0 +1,241 @@
+     * and was passed an index, as soon as it can it will try to show the alert that would edit
+     * the item at that index.
+     * context is the context to which you want to open this sheet in.
+     * layerNames is the list of strings you want to appear in the bottom sheet as well as check against during the alert dialog.
+     * wasAlertOpen is to determine if the alert was open previous to this call and if it was to try to open it.
+     * activeEditIndex pairs with wasAlertOpen to determine which item in the list to edit, if not specified no alert will be shown
+     * even if wasAlertOpen is set to true.
+     * onAlertDialogOpenState, invoked with true when the alert dialog of an edit is show, invokes with false when the dialog is closed.
+     * onAlertDialogTextChange, passed to the showLayerAlertDialog as its onTextChanged lambda.
+     * onEdit, invoked with the index of the layer being edited before the alert dialog is shown.
+     * onDismiss, invoked on any dismissal of the bottom sheet.
+     */
+    fun showViewingLayersDialog(
+        context: Context,
+        layerNames: MutableList<String>,
+        alertDialogRefText: String = "",
+        wasAlertOpen: Boolean = false,
+        activeEditIndex: Int = -1,
+        /**
+         * used to set the value of the boolean used to determine whether the alert dialog was open
+         * invoked on the start of an edit with true, and invoked with false on dismissal of the edit alert dialog
+         * ideally setting the state of the boolean to true on the opening of the dialog and false on the closing
+         */
+        onAlertDialogOpenState: (Boolean) -> Unit,
+        /**
+         * passed to the showLayerAlertDialog function on the edit call as its onTextChanged lambda
+         */
+        onAlertDialogTextChange: (String) -> Unit, //handles text change
+        /**
+         * used to assign the index that we are actively editing
+         * Ideally hold an outside variable that we update useing the int with this
+         * pass that variable as the activeEditIndex
+         */
+        onEdit: (Int) -> Unit,
+        /**
+         * used to determine whether we have the sheet open or not
+         * ideally sets a boolean that tracks the open state to false
+         * note: unlike the alert dialog it does not need a onOpen to set true, can simply do that before function call
+         */
+        onDismiss: () -> Unit
+    ): BottomSheetDialog {
+        val data = mutableListOf<LayerViewModel>()
+        for (x in layerNames) data.add(LayerViewModel(x))
+
+        lateinit var myAdapter: EditLayerAdapter
+        myAdapter = EditLayerAdapter(
+            data = data,
+            onDeleteClick = { layerModel ->
+                val index = data.indexOf(layerModel)
+                data.removeAt(index)
+                myAdapter.notifyItemRemoved(index)
+            },
+            onEditClick = { layerModel ->
```

Comment:
I did a lot of this logic managing the layerList in my viewModel, but I guess I didn't consider doing it in the custom class for this.

I don't know if this is "wrong", it just strikes me as a distinctly different choice. All of this is related to view level logic, and I see context being passed around a lot, so I don't know if reworking this into a viewModel would even be worthwhile at this point

Curious what Nic will have on this one